### PR TITLE
Move all query service HTTP handlers into one function

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?
- Prequel for #6121

## Description of the changes
- Consolidate business logic handlers into a single function `initRouter`
- Move tenancy handling to the same function, remove it from API and APIv3 handlers

## How was this change tested?
- `go test ./cmd/query/app/...`

### CODE REVIEW

**Comment by yurishkuro** [comment]
@mahadzaryab1 please stamp

File: cmd/query/app/server.go [hunk 1]
```diff
@@ -228,17 +226,34 @@ func createHTTPServer(
...
+
 	var handler http.Handler = r
-	handler = responseHeadersHandler(handler, queryOpts.HTTP.ResponseHeaders)
 	if queryOpts.BearerTokenPropagation {
 		handler = bearertoken.PropagationHandler(telset.Logger, handler)
 	}
+	if tenancyMgr.Enabled {
+		handler = tenancy.ExtractTenantHTTPHandler(tenancyMgr, handler)
+	}
+	// TODO add tenancy handler
```
Comment (mahadzaryab1) [inline cmd/query/app/server.go:243]: i see the tenancy handler above so we can remove this comment

File: cmd/query/app/server.go [hunk 2]
```diff
@@ -249,18 +264,17 @@ func createHTTPServer(
...
 			ReadHeaderTimeout: 2 * time.Second,
 		},
+		staticHandlerCloser: staticHandlerCloser,
 	}
 
 	if queryOpts.HTTP.TLSSetting != nil {
 		tlsCfg, err := queryOpts.HTTP.TLSSetting.LoadTLSConfig(ctx) // This checks if the certificates are correctly provided
 		if err != nil {
-			return nil, err
+			return nil, errors.Join(err, staticHandlerCloser.Close())
```
Comment (mahadzaryab1) [inline cmd/query/app/server.go:273]: ah that's why I was seeing the goleak errors - need to close the handler before exiting

File: cmd/query/app/server.go [hunk 3]
```diff
@@ -203,14 +203,13 @@ type httpServer struct {
 
 var _ io.Closer = (*httpServer)(nil)
 
-func createHTTPServer(
-	ctx context.Context,
+func initRouter(
```
Comment (mahadzaryab1) [inline cmd/query/app/server.go:206]: will it be okay to have redundant handlers with the otel implementation? specifically asking about the response headers handler and the compression handler.

**Review by mahadzaryab1 — APPROVED** [review]
lgtm

File: cmd/query/app/server.go (hunk 3 above)
Comment (yurishkuro) [reply cmd/query/app/server.go:206]: yes

**Pull request merged**